### PR TITLE
LOS-5416: Create button base and dot spinner components

### PR DESCRIPTION
**Pull request by rs-rahul**
## Description
- Create a button base component
- Create a loading spinner
- Add missing colors in tailwind config
- Add font-awesome
- Add sizing to button
- Add icons to buttons

## JIRA Link
https://firsthelpfinancial.atlassian.net/browse/LOS-5416

## Screenshot
image

### CODE REVIEW

File: src/components/Button/Button.tsx [hunk 1]
```diff
@@ -1,25 +1,84 @@
...
+        {
+          'text-blue-grey-500 bg-blue-grey-700 pointer-events-none': disabled,
+          'bg-aqua-100 px-12': loading,
+        }
+      );
+
+    case 'icon':
+      return classNames('border-0 bg-primary text-white hover:bg-blue-400 focus:bg-blue-400 active:bg-blue-400', {
+        'text-blue-500 bg-blue-grey-700 pointer-events-none': disabled,
+      });
```
Comment (robin-srs) [inline src/components/Button/Button.tsx:61]: lets extract these to a css file

File: src/components/Button/Button.tsx [hunk 2]
```diff
@@ -1,25 +1,84 @@
...
+        classNames(
+          'font-noto-sans font-bold px-5 py-2 text-base rounded-full flex gap-2 leading-6 border',
+          {
+            'pointer-events-none': disabled,
+            'px-8': loading,
+          },
+          className
+        )
+      )}>
+      {!loading ? children : <DotSpinner circleClassName="bg-current" radius={12} />}
```
Comment (robin-srs) [inline src/components/Button/Button.tsx:27]: can we provide a loading spinner (react node) through props

File: src/components/Button/Button.tsx [hunk 3]
```diff
@@ -1,25 +1,84 @@
...
+    return <ButtonBase {...otherProps}>{children}</ButtonBase>;
+  }
 
-  return <ButtonBase {...otherProps}>{children}</ButtonBase>;
+  return (
+    <ButtonBase
+      {...otherProps}
+      loading={loading}
+      disabled={disabled}
+      className={classNames(getStyles({ variant, size, disabled, loading }), className)}>
```
Comment (robin-srs) [inline src/components/Button/Button.tsx:80]: is size prop implementation still left?

File: src/theme/colors.ts [hunk 4]
```diff
@@ -1,4 +1,10 @@
 export const PALETTE_COLORS = {
+  transparent: 'transparent',
```
Comment (robin-srs) [inline src/theme/colors.ts:2]: i think transparent and current already exists in tailwind

**Review by sujal7 — APPROVED** [review]

**Review by robin-srs — COMMENTED** [review]
import the spinner in the components indext.ts so that it can be shipped as well

File: src/theme/colors.ts [hunk 5]
```diff
@@ -1,4 +1,8 @@
 export const PALETTE_COLORS = {
+  primary: '#00497B', // Regal Blue
+  secondary: '#F00', // Red
+  tertiary: '#038789', // Blue Lagoon
+  // transparent: 'transparent',
```
Comment (robin-srs) [inline src/theme/colors.ts:5]: remove comments

**Review by robin-srs — APPROVED** [review]

**Pull request merged**